### ISSUE TITLE
Add MacOS ARM support or fail cleanly

### ISSUE THREAD

**Issue by pradyunsg** [a-hurst/pysdl2-dll#14]
```
[venv]/lib/python3.9/site-packages/sdl2/dll.py:213: DLLWarning: OSError("dlopen([venv]/lib/python3.9/site-packages/sdl2dll/dll/SDL2_image.framework/Versions/A/SDL2_image, 0x0006): tried: '[venv]/lib/python3.9/site-packages/sdl2dll/dll/SDL2_image.framework/Versions/A/SDL2_image' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' (no such file), '/System/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' (no such file)")
```

The relevant bit is:

> `(have 'x86_64', need 'arm64e')`

Which is likely because the libraries for ARM on MacOS are not properly compiled/available?

**Comment by a-hurst** [comment]
@pradyunsg Yep, I'm currently waiting on new official libSDL binaries so I can give pysdl2-dll full Apple Silicon support, since modern macOS won't run unsigned binaries from the internet and I don't have a paid developer account to sign them myself 😕 SDL2 and `SDL2_ttf` are both native already, so I'm just waiting on new releases of `SDL2_mixer` and `SDL_image`.

You're absolutely right that this needs better and more informative error handling though. I guess I was hoping we'd get official binaries sooner and that error handling would quickly become unnecessary, but at this point I suppose I've left it too long.

In terms of implementation, I guess the easiest option would be to check for arm64 macOS during the DLL fetch script and skip, treating Apple Silicon like other unsupported platforms/architectures unless we're running Python through Rosetta 2. That said, I'm not sure if pip suppresses `print()` warning messages during installation to let users know what's going on.

**Comment by pradyunsg** [comment]
If you error out in `setup.py bdist_wheel` (instead of building a wheel with a MacOS tag), the error output will be printed by pip. If the build succeeds, pip won't print any output unless you pass `--verbose/-v` to it.

**Comment by tsedan** [comment]
Are there status updates on this @a-hurst? Are you still waiting for the official SDL2_mixer and SDL_image binaries?

**Comment by a-hurst** [comment]
@tsedan SDL *just* released pre-release mixer and image binaries for macOS ARM a few days ago! I've got some other obligations on my time this week, but I'm hoping to have an updated ARM-native pysdl2-dll up by the end of the month at the latest.

Also, `SDL2_gfx` unfortunately doesn't have any signed macOS binaries (for Intel or Apple Silicon) so until I find a contributor with a developer account willing to sign those that'll still be a GateKeeper annoyance, but no more so for ARM than Intel.

**Comment by a-hurst** [comment]
Closed by #16 and #17, release coming shortly!
